### DISCUSSION TITLE
typescript로 마이그레이션 시 useContext에서의 type 지정

### DISCUSSION THREAD

**Discussion by JiyeonP** [wafflestudio/seminar-2021#642] [Q&A] [answered]
우선 아래와 같이 useAuthContext를 구현한 상태입니다.

![image](https://user-images.githubusercontent.com/66158942/140533685-5dba87b3-eb7f-4231-9087-a154aa099ea7.png)

이 중 tokenExpire를 받아오려고 했으나, 아래와 같은 에러가 발생하였습니다.
![image](https://user-images.githubusercontent.com/66158942/140533993-d2c1c5e1-2c50-41c0-b87f-de22ac408e65.png)

() => void; 를 type으로 작성하고 { tokenExpire } = useAuthContext<타입>();으로 작성하였더니 'expected 0 argument, but got 1'이라는 에러가 발생하였습니다.

context를 가져올 때에는 type 지정을 어떻게 해야하나요?

**Comment by woohm402** [comment, marked as answer]
저는 보통 export 할 때

```ts
type AuthContextType = {
  login: boolean;
  등등
}

export const useAuthContext = () => useContext(AuthContext) as AuthContextType;
```

이런 식으로 export해 줍니다!

**Reply by JiyeonP** [reply]
Context 파일도 typescript로마이그레이션 해야 하는건가요?

**Reply by woohm402** [reply]
그게 제일 깔끔한 방법이긴 합니다

사실 그냥 해당 컴포넌트에서

```ts
const { tokenExpire } = useAuthContext();
```

이걸

```ts
const { tokenExpire } = useAuthContext() as AuthContextType;
```

로 변경하면 어떻게 해결은 되겠으나 그렇게 추천드리고 싶은 방법은 아닙니다 :)

**Reply by JiyeonP** [reply]
혹시 이 에러는 어떻게 해결해야 하는지 도움 주실 수 있으실까요?

![image](https://user-images.githubusercontent.com/66158942/140538666-69e514a4-506e-422c-ba22-40ab9263ee91.png)

**Reply by woohm402** [reply]
이런 경우는 ts-ignore를 처리할만할 것 같습니다

사실 타스 쓰는 이유가 실수 안 하려고 쓰는 건데 저런 거까지 일일이 타입 잡아줄 필요가 없으니까요!

근데 전 저런 경험을 해본 적이 없는데.. 처음 보는 오류네요 😶

**Reply by JiyeonP** [reply]
감사합니다!
